pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -1127,6 +1127,38 @@ func TestValidatePipelineParameterVariables_Success(t *testing.T) {
 				Name: "a-param-intended-to-be-array", Value: ParamValue{Type: ParamTypeString, StringVal: "$(params.myArray[*])"},
 			}},
 		}},
+	}, {
+		name: "valid array parameter variables in matrix",
+		params: []ParamSpec{{
+			Name: "baz", Type: ParamTypeArray, Default: &ParamValue{Type: ParamTypeArray, ArrayVal: []string{"some", "default"}},
+		}, {
+			Name: "foo-is-baz", Type: ParamTypeArray,
+		}},
+		tasks: []PipelineTask{{
+			Name:    "bar",
+			TaskRef: &TaskRef{Name: "bar-task"},
+			Matrix: &Matrix{
+				Include: []MatrixInclude{{
+					Name: "build-1",
+					Params: []Param{{
```

Comment:
can this test case be simplified? (e.g. fewer combinations)

Reply:
Simplified